python/pyarrow/parquet.py

```diff
@@ -1271,7 +1391,41 @@ def _make_manifest(path_or_paths, fs, pathsep='/', metadata_nthreads=1,
 def read_table(source, columns=None, use_threads=True, metadata=None,
                use_pandas_metadata=False, memory_map=False,
                read_dictionary=None, filesystem=None, filters=None,
-               buffer_size=0):
+               buffer_size=0, use_dataset=False):
+    if use_dataset:
+        import pyarrow.dataset as ds
+        import pyarrow.fs
+
+        # map old filesystems to new one
+        if isinstance(filesystem, LocalFileSystem):
+            filesystem = pyarrow.fs.LocalFileSystem()
+
+        # map additional arguments
+        # TODO raise warning when unsupported arguments are passed
+        reader_options = {}
+        if buffer_size:
+            reader_options.update(use_buffered_stream=True,
+                                  buffer_size=buffer_size)
+        if read_dictionary is not None:
+            reader_options.update(dict_columns=read_dictionary)
+        parquat_format = ds.ParquetFileFormat(reader_options=reader_options)
+
+        dataset = ds.dataset(source, filesystem=filesystem,
+                             format=parquat_format, partitioning="hive")
+        if filters is not None and not isinstance(filters, ds.Expression):
+            filters = _filters_to_expression(filters)
```

Comment:
This seems to be duplicated in ParquetDatasetV2.__init__, maybe extract _dataset_from_legacy_args()?

Reply:
Moved this into a function to deduplicate